File: Examples/CSharp/WorkingWithCalendars/RetrieveCalendarInfo.cs
namespace Aspose.Tasks.Examples.CSharp.WorkingWithCalendars
{
    using System;

    internal class RetrieveCalendarInfo
    {
        public static void Run()
        {
            // The path to the documents directory.
            var dataDir = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName);

            //ExStart:RetrieveCalendarInfo
            //ExFor: Calendar.Uid
            //ExFor: Calendar.Name
            //ExFor: Calendar.WeekDays
            //ExSummary: Shows how to retrieve calendar info.
            // Create a project instance
            var project = new Project(dataDir + "RetrieveCalendarInfo.mpp");

            // Retrieve Calendars Information
            foreach (var calendar in project.Calendars)
            {
                if (calendar.Name == null)
                {
                    continue;
                }

                Console.WriteLine("Calendar UID: " + calendar.Uid);
                Console.WriteLine("Calendar Name: " + calendar.Name);

                foreach (var weekDay in calendar.WeekDays)
                {
                    if (!weekDay.DayWorking) 
                    {
                        continue;
                    }

                    var workingTime = weekDay.GetWorkingTime();
                    Console.Write(weekDay.DayType + ": ");
                    Console.WriteLine(workingTime.ToString());
                }
            }
            //ExEnd:RetrieveCalendarInfo
        }
    }
}

File: Examples/VisualBasic/WorkingWithCalendars/RetrieveCalendarInfo.vb
Imports System
Imports Aspose.Tasks

'
'This project uses Automatic Package Restore feature of NuGet to resolve Aspose.Tasks for .NET API reference 
'when the project is build. Please check https://docs.nuget.org/consume/nuget-faq for more information. 
'If you do not wish to use NuGet, you can manually download Aspose.Tasks for .NET API from http://www.aspose.com/downloads, 
'install it and then add its reference to this project. For any issues, questions or suggestions 
'please feel free to contact us using http://www.aspose.com/community/forums/default.aspx
'

Namespace WorkingWithCalendars
    Public Class RetrieveCalendarInfo
        Public Shared Sub Run()
            ' ExStart: RetrieveCalendarInfo
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType.FullName)

            ' Create a project reader instance
            Dim project As New Project(dataDir & "RetrieveCalendarInfo.mpp")

            ' Retrieve Calendars Information
            Dim alCals As CalendarCollection = project.Calendars
            For Each cal As Calendar In alCals
                If cal.Name IsNot Nothing Then
                    Console.WriteLine("Calendar UID : " & cal.Uid)
                    Console.WriteLine("Calendar Name : " & cal.Name)

                    Dim alDays As WeekDayCollection = cal.WeekDays
                    For Each wd As WeekDay In alDays
                        Dim ts As TimeSpan = wd.GetWorkingTime()
                        If wd.DayWorking Then
                            Console.WriteLine(wd.DayType.ToString() & ":")
                            Console.WriteLine(ts.ToString())
                        End If
                    Next wd
                End If
            Next cal
            ' ExEnd: RetrieveCalendarInfo
        End Sub
    End Class
End Namespace